R/ipc_get.R

```diff
@@ -43,36 +49,34 @@ ipc_get <- function(
       )
     }
 
-    ipc_list <- httr::content(
-      x = resp,
-      as = "parsed",
-      type = "application/json"
-    )
-
-    # check if list of lists or if not, convert to a list of single list
-    if (!all(sapply(ipc_list, inherits, what = "list"))) {
-      ipc_list <- list(ipc_list)
-    }
+    # check that the expected return value is correct
+    expected_format <- ifelse(return_format == "csv", "text/csv", "application/json")
+    resp_format <- httr::headers(resp)$`content-type`
 
-    if (purrr::is_empty(ipc_list)) {
+    if (resp_format != expected_format) {
       stop(
-        "API call was successful but no content was returned. Check that the ",
-        "ID or other parameters passed are actually present in the resource.",
+        "API returned '",
+        resp_format,
+        "'not the expected '",
+        expected_format,
+        "'. Check the API and consider raising an issue on the {ripc} GitHub.",
         call. = FALSE
```

Comment:
Hmm, possibly, but how did you run `ipc_get_population()` with `return_format = "csv"), because that's not actually possible unless you're editing the internals of the function. I have explicit arg catching in `ipc_get_areas()`, which is the only place that people can explicitly change the return format to get a `geojson` file. The user would need to use `ripc:::ipc_get()` manually to be able to incorrectly pass something, so I think for now can leave it unchecked.